File: app/src/main/java/id/ac/polinema/seameo/ecanteen/view/activity/HomeActivity.java
/*
 * Copyright (c) 2019.
 *
 * Author: Mahatta Maulana
 * Github: https://github.com/hattamaulana
 *
 * Last Modified at 7/2/19 4:36 PM
 */

package id.ac.polinema.seameo.ecanteen.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import java.util.ArrayList;

import id.ac.polinema.seameo.ecanteen.R;
import id.ac.polinema.seameo.ecanteen.contract.TransactionContract;
import id.ac.polinema.seameo.ecanteen.model.TransactionModel;
import id.ac.polinema.seameo.ecanteen.presenter.transaction.TransactionPresenter;

public class HomeActivity extends AppCompatActivity implements TransactionContract.View {
    private TransactionPresenter mPresenter;
    private ArrayList<TransactionModel> mListHistory;

    @Override
    public void initPresenter() {
        mPresenter = new TransactionPresenter();
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        initPresenter();
        initView();
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    private void initView() {
        FloatingActionButton btn = findViewById(R.id.scann_barcode);
        btn.setOnClickListener(onClickFAButton());
    }

    private View.OnClickListener onClickFAButton() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(HomeActivity.this, ScanActivity.class));
            }
        };
    }

}

File: app/src/main/java/id/ac/polinema/seameo/ecanteen/view/activity/HomeActivity.kt
/*
 * Copyright (c) 2019.
 *
 * Author: Mahatta Maulana
 * Github: https://github.com/hattamaulana
 *
 * Last Modified at 9/27/19 9:00 PM
 */

package id.ac.polinema.seameo.ecanteen.view.activity

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.fragment.NavHostFragment
import id.ac.polinema.seameo.ecanteen.R
import id.ac.polinema.seameo.ecanteen.SharedPref
import id.ac.polinema.seameo.ecanteen.view.utils.setWindow
import java.util.*

class HomeActivity : AppCompatActivity() {
    val TAG = this.javaClass.simpleName

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Set status bar transparent
        // And set content view
        setWindow(window)
        setContentView(R.layout.activity_home)

        val navHost = supportFragmentManager
                .findFragmentById(R.id.homeFragment) as NavHostFragment
        val navController = navHost.navController

        navController.addOnDestinationChangedListener { _, destination, _ ->
            val currentDest = resources.getResourceName(destination.id)
            Log.i(TAG, "Current Destination : $currentDest")
        }
    }

    override fun onStart() {
        super.onStart()
        val sharedPref = SharedPref(application)

        if (sharedPref.uid == "") {
            Log.i(TAG, "onStart() : Set SharedPreferences")

            sharedPref.uid = UUID.randomUUID().toString();
        }
    }
}
